reconcile/utils/gql.py

```diff
@@ -218,7 +231,28 @@ def commit_timestamp_utc(self) -> Optional[str]:
         return None
 
 
-_gqlapi: Optional[GqlApi] = None
+class GqlApiSingleton:
+    gql_api: Optional[GqlApi] = None
+    gqlapi_lock = threading.Lock()
+
+    def __new__(cls, *args, **kwargs):
+        with cls.gqlapi_lock:
+            if not cls.gql_api:
+                cls.gql_api = GqlApi(*args, **kwargs)
+        return cls
```

Comment:
usually `__new__` returns instance object

Reply:
Will adjust

Reply:
__new__ should return the instance of the created class. returning another type might work, but is very counterintuitive. Also mypy complains :-)

Reply:
How about not using magic methods, just plain old classmethod to make it explicit

```
    @classmethod
    def init(cls, *args, **kwargs) -> GqlApi:
        with cls.gqlapi_lock:
            if cls.gql_api:
                logging.debug("Resestting GqlApi instance")
                cls.close()
            cls.gql_api = GqlApi(*args, **kwargs)
            return cls.gql_api
```